rdf-abac-core/src/main/java/io/telicent/jena/abac/assembler/AttributeStoreBuildLib.java

```diff
@@ -143,4 +163,34 @@ private static String lookupEnvironmentVariable(String name) {
             throw new AssemblerException(root, "Failed to parse the attributes store file '"+attributesStoreFilename+"'", th);
         }
     }
+
+    /*package*/ static Duration parseDurationProperty(Resource root, Property property, Duration defaultDuration) {
+        String propertyAsString = GraphUtils.getStringValue(root, property);
```

Comment:
This is the object, not the property.

Reply:
updated

Reply:
It does not check the datatype.
Given this is related to security, maybe it is better to be quite careful.

Add in this class because GraphUtils doe snot have it (yet!):
```
    public static String getStringValue(Resource r, Property p, RDFDatatype datatype) {
        RDFNode obj = getAsRDFNode(r, p);
        if ( obj == null )
            return null;
        Literal literal = obj.asLiteral();
        if ( ! datatype.equals(literal.getDatatype()) )
            throw new JenaException("Not the expected datatype2");
        return obj.asLiteral().getString();
    }
```